cmd/metacache-entries.go

```diff
@@ -481,7 +484,12 @@ func (m *metaCacheEntriesSorted) fileInfoVersions(bucket, prefix, delimiter, aft
 
 			for _, version := range fiVersions {
 				versioned := vcfg != nil && vcfg.Versioned(entry.name)
-				versions = append(versions, version.ToObjectInfo(bucket, entry.name, versioned))
+				oi := version.ToObjectInfo(bucket, entry.name, versioned)
+				versions = append(versions, oi)
+				queueReplicationHeal(ctx, bucket, oi, replicationConfig{
+					Config:  rcfg,
+					remotes: tgts,
+				})
```

Comment:
wrong place to queue this @poornas you should queue this where the lifecycle gets queued.

Reply:
fixed